### PR TITLE
Remove use of unit IDs for transports

### PR DESCRIPTION
**Pull request by sixlettervariables**
This removes the use of unit IDs for "transports" and cleans up how we tracked where units were being transported. Previously we used a Map data structure, except it could only have one entry. This has been instead switched out for a more formal class, `TransportShipAssignment`. Tests have been added for the basic transport logic as well.

### CODE REVIEW

**Comment by Windchild292** [comment]
Started to look over it and it got me thinking: is there any chance that it was a map in preparation for units assigned to units assigned to units? For example, infantry assigned to an APC assigned to a DropShip?

**Comment by sixlettervariables** [comment]
The map tracked who contained the unit and in which bay. Having multiple entries would be confusing as you would not know which actually contained the unit without enumerating the map and checking the keys (transports) to see which contained the unit (getTransportedUnits).

File: MekHQ/src/mekhq/campaign/unit/Unit.java [hunk 1]
```diff
@@ -1569,20 +1611,23 @@ public double calcInfantryBayWeight(Entity unit) {
...
-        }
-        for (UUID id : u.getTransportShipId().keySet()) {
-            int bayNumber = u.getTransportShipId().get(id);
-            updateBayCapacity(unitType, unitWeight, true, bayNumber);
-        }
-        removeTransportedUnit(u.getId());
-        u.getTransportShipId().clear();
+        // Only unload and update the transport assignment for a unit
+        // if we have actually transported them.
+        if (removeTransportedUnit(u)) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/unit/Unit.java:1616]: What about the case whereby a unit is assigned to this transport according to the unit but not according to the transport ship?

File: MekHQ/src/mekhq/campaign/unit/Unit.java [hunk 2]
```diff
@@ -1569,20 +1611,23 @@ public double calcInfantryBayWeight(Entity unit) {
...
+            double unitWeight;
+            if (u.getEntity().getUnitType() == UnitType.INFANTRY) {
+                unitWeight = calcInfantryBayWeight(u.getEntity());
+            } else {
+                unitWeight = u.getEntity().getWeight();
+            }
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/unit/Unit.java:1622]: Trinary candidate.

**Review by Windchild292 — COMMENTED** [review]
A few minor things first

**Comment by sixlettervariables** [comment]
I've added the fix for #1801 to this as well.

**Review by Windchild292 — APPROVED** [review]
Just wanted to check that there isn't any logic in TOEMouseAdapter that should be removed given the logic movement for #1801? Otherwise, approved.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**